File: tests/fixtures/vulnerable/supply-chain.py
# INTENTIONALLY VULNERABLE — Sentinel rule-tester fixtures
# DO NOT use in production. Every pattern here is a known vulnerability.
# CWE-502, CWE-22, CWE-94, CWE-829, CWE-506

import pickle
import yaml
import os
import zipfile
import tarfile

# CWE-502: Pickle deserialization from untrusted source
def load_model(model_path):
    with open(model_path, 'rb') as f:
        model = pickle.load(f)
    return model

# CWE-502: YAML load without SafeLoader
def parse_pipeline_config(config_str):
    config = yaml.load(config_str)
    return config

# CWE-502: jsonpickle (unsafe by default)
import jsonpickle
def deserialize_task(data):
    return jsonpickle.decode(data)

# CWE-22: Zip slip — extracting without path validation
def extract_package(zip_path, dest_dir):
    with zipfile.ZipFile(zip_path) as zf:
        zf.extractall(dest_dir)

# CWE-22: Tar slip
def extract_tarball(tar_path, dest_dir):
    with tarfile.open(tar_path) as tf:
        tf.extractall(dest_dir)

# CWE-94: Dynamic import from user-specified module
def load_plugin(plugin_name):
    module = __import__(plugin_name)
    return module.run()

# CWE-94: urllib.request.urlopen with user URL
def fetch_remote(url):
    import urllib.request
    data = urllib.request.urlopen(url).read()
    return data

# CWE-829: Loading from untrusted CDN without SRI
SCRIPT_TAG = '<script src="http://cdn.example.com/lib.js"></script>'

# CWE-502: shelve (uses pickle internally)
import shelve
def load_cache():
    db = shelve.open('cache.db')
    data = db['user_prefs']
    return data

File: tests/fixtures/vulnerable/supply-chain.php
<?php
// INTENTIONALLY VULNERABLE — Sentinel rule-tester fixtures
// DO NOT use in production. Every pattern here is a known vulnerability.
// CWE-502, CWE-94, CWE-22

// CWE-502: Unsafe unserialize from user input
$data = unserialize($_POST['data']);
echo $data->name;

// CWE-22: Path traversal via user input
$file = $_GET['file'];
include('/templates/' . $file);

// CWE-20: Direct use of superglobals without filtering
$user_id = $_GET['id'];
$query = "SELECT * FROM users WHERE id = " . $user_id;
$result = mysqli_query($conn, $query);

// CWE-502: Session handling
$_SESSION['prefs'] = $_POST['preferences'];
session_start();

// CWE-22: File inclusion
$page = $_GET['page'];
require_once($page . '.php');

// Access superglobals directly without sanitization
$cookie_val = $_COOKIE['session_token'];
$server_val = $_SERVER['HTTP_REFERER'];
echo $server_val;

// Session without regeneration
session_start();
$_SESSION['user'] = $_POST['username'];
$_SESSION['role'] = $_GET['role'];
?>
